diff -up firefox-52.0b8/js/src/jsmath.cpp.missing-getrandom firefox-52.0b8/js/src/jsmath.cpp
--- firefox-52.0b8/js/src/jsmath.cpp.missing-getrandom	2017-02-22 08:09:21.764353407 +0100
+++ firefox-52.0b8/js/src/jsmath.cpp	2017-02-22 08:10:36.939330775 +0100
@@ -78,7 +78,7 @@
 static_assert(GETRANDOM_NR == SYS_getrandom,
               "GETRANDOM_NR should match the actual SYS_getrandom value");
 #  endif
-# else
+# elif defined(GETRANDOM_NR)
 #  define SYS_getrandom GETRANDOM_NR
 # endif
 
@@ -721,7 +721,7 @@ js::GenerateRandomSeed()
     seed = (static_cast<uint64_t>(arc4random()) << 32) | arc4random();
 #elif defined(XP_UNIX)
     bool done = false;
-# if defined(__linux__)
+# if defined(__linux__) && defined(SYS_getrandom)
     // Try the relatively new getrandom syscall first. It's the preferred way
     // on Linux as /dev/urandom may not work inside chroots and is harder to
     // sandbox (see bug 995069).
