diff -up mozilla-aurora/js/src/jit/AtomicOperations.h.s390-atomic mozilla-aurora/js/src/jit/AtomicOperations.h
--- mozilla-aurora/js/src/jit/AtomicOperations.h.s390-atomic	2016-11-15 12:22:38.000000000 +0100
+++ mozilla-aurora/js/src/jit/AtomicOperations.h	2016-12-08 15:36:11.792439891 +0100
@@ -341,6 +341,9 @@ AtomicOperations::isLockfree(int32_t siz
 #  include "jit/none/AtomicOperations-ppc.h"
 # elif defined(__aarch64__)
 #  include "jit/arm64/AtomicOperations-arm64.h"
+# elif defined(__s390__)
+   // Also works with ppc atomic operations
+#  include "jit/none/AtomicOperations-ppc.h"
 # else
 #  include "jit/none/AtomicOperations-none.h" // These MOZ_CRASH() always
 # endif
