diff -up firefox-52.0esr/security/certverifier/CTLogVerifier.cpp.nss-prbool firefox-52.0esr/security/certverifier/CTLogVerifier.cpp
--- firefox-52.0esr/security/certverifier/CTLogVerifier.cpp.nss-prbool	2017-02-28 02:56:03.000000000 +0100
+++ firefox-52.0esr/security/certverifier/CTLogVerifier.cpp	2017-03-01 11:34:15.623577628 +0100
@@ -7,6 +7,7 @@
 #include "CTLogVerifier.h"
 
 #include "CTSerialization.h"
+#include "prtypes.h"
 #include "hasht.h"
 #include "mozilla/ArrayUtils.h"
 #include "mozilla/Assertions.h"
diff -up firefox-52.0esr/security/manager/ssl/nsSSLStatus.cpp.nss-prbool firefox-52.0esr/security/manager/ssl/nsSSLStatus.cpp
--- firefox-52.0esr/security/manager/ssl/nsSSLStatus.cpp.nss-prbool	2017-03-01 12:09:16.649945106 +0100
+++ firefox-52.0esr/security/manager/ssl/nsSSLStatus.cpp	2017-03-01 12:09:35.749939355 +0100
@@ -4,6 +4,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "prtypes.h"
 #include "mozilla/Casting.h"
 #include "nsSSLStatus.h"
 #include "nsIClassInfoImpl.h"
