diff -up firefox-52.1.0esr/dom/filesystem/compat/FileSystemDirectoryReader.h.s390-build-includes-fix firefox-52.1.0esr/dom/filesystem/compat/FileSystemDirectoryReader.h
--- firefox-52.1.0esr/dom/filesystem/compat/FileSystemDirectoryReader.h.s390-build-includes-fix	2017-04-12 09:31:39.419402551 +0200
+++ firefox-52.1.0esr/dom/filesystem/compat/FileSystemDirectoryReader.h	2017-04-12 09:31:53.392340589 +0200
@@ -11,6 +11,7 @@
 #include "mozilla/ErrorResult.h"
 #include "mozilla/dom/BindingDeclarations.h"
 #include "nsCycleCollectionParticipant.h"
+#include "mozilla/dom/FileSystemDirectoryEntry.h"
 #include "nsWrapperCache.h"
 
 namespace mozilla {
