diff -up mozilla-esr17/browser/app/profile/firefox.js.addons mozilla-esr17/browser/app/profile/firefox.js
--- mozilla-esr17/browser/app/profile/firefox.js.addons	2012-11-29 05:43:51.000000000 +0100
+++ mozilla-esr17/browser/app/profile/firefox.js	2012-12-21 08:40:51.799335531 +0100
@@ -61,7 +61,7 @@ pref("extensions.hotfix.certs.1.sha1Fing
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.jsm for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
 
 // Dictionary download preference
 pref("browser.dictionaries.download.url", "https://addons.mozilla.org/%LOCALE%/firefox/dictionaries/");
diff -up mozilla-esr17/toolkit/mozapps/extensions/XPIProvider.jsm.addons mozilla-esr17/toolkit/mozapps/extensions/XPIProvider.jsm
--- mozilla-esr17/toolkit/mozapps/extensions/XPIProvider.jsm.addons	2012-11-29 05:45:08.000000000 +0100
+++ mozilla-esr17/toolkit/mozapps/extensions/XPIProvider.jsm	2012-12-21 08:40:14.342341623 +0100
@@ -1630,8 +1630,8 @@ var XPIProvider = {
       this.showUpgradeUI();
       flushCaches = true;
     }
-    else if (aAppChanged === undefined) {
-      // For new profiles we will never need to show the add-on selection UI
+    else {
+      // For all profiles we will never need to show the add-on selection UI
       Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
     }
 
