diff -up mozilla-release/browser/installer/package-manifest.in.old mozilla-release/browser/installer/package-manifest.in
--- mozilla-release/browser/installer/package-manifest.in.old	2015-04-03 04:30:05.000000000 +0200
+++ mozilla-release/browser/installer/package-manifest.in	2015-04-07 13:17:44.180343700 +0200
@@ -608,7 +608,7 @@
 @RESPATH@/components/MozKeyboard.js
 @RESPATH@/components/InputMethod.manifest
 
-#ifdef MOZ_DEBUG
+#ifdef ENABLE_TESTS
 @RESPATH@/components/TestInterfaceJS.js
 @RESPATH@/components/TestInterfaceJS.manifest
 #endif
diff -up firefox-38.0/mozilla-beta/js/src/jit/shared/CodeGenerator-shared-inl.h.debug firefox-38.0/mozilla-beta/js/src/jit/shared/CodeGenerator-shared-inl.h
--- mozilla-beta/js/src/jit/shared/CodeGenerator-shared-inl.h.debug	2015-03-31 00:54:47.000000000 +0200
+++ mozilla-beta/js/src/jit/shared/CodeGenerator-shared-inl.h	2015-04-07 21:36:41.136826585 +0200
@@ -189,7 +189,7 @@ CodeGeneratorShared::verifyHeapAccessDis
                                                  Scalar::Type type, const Operand& mem,
                                                  LAllocation alloc)
 {
-#ifdef DEBUG
+#if 0
     using namespace Disassembler;
 
     OtherOperand op;
