diff -up firefox-31.1.0/mozilla-esr31/browser/app/profile/firefox.js.plugins firefox-31.1.0/mozilla-esr31/browser/app/profile/firefox.js
--- firefox-31.1.0/mozilla-esr31/browser/app/profile/firefox.js.plugins	2014-09-11 13:47:45.000000000 +0200
+++ firefox-31.1.0/mozilla-esr31/browser/app/profile/firefox.js	2014-09-11 14:42:26.036810267 +0200
@@ -657,7 +657,7 @@ pref("plugins.click_to_play", true);
 
 pref("plugins.hideMissingPluginsNotification", false);
 
-pref("plugin.default.state", 1);
+pref("plugin.default.state", 2);
 
 // Plugins bundled in XPIs are enabled by default.
 pref("plugin.defaultXpi.state", 2);
@@ -665,7 +665,7 @@ pref("plugin.defaultXpi.state", 2);
 // Flash is enabled by default, and Java is click-to-activate by default on
 // all channels.
 pref("plugin.state.flash", 2);
-pref("plugin.state.java", 1);
+pref("plugin.state.java", 2);
 
 // Whitelist Requests
 
