diff -up xulrunner-17.0/mozilla-esr17/media/webrtc/trunk/src/typedefs.h.old xulrunner-17.0/mozilla-esr17/media/webrtc/trunk/src/typedefs.h
--- xulrunner-17.0/mozilla-esr17/media/webrtc/trunk/webrtc/typedefs.h.old	2012-11-16 20:47:22.000000000 +0100
+++ xulrunner-17.0/mozilla-esr17/media/webrtc/trunk/webrtc/typedefs.h	2012-11-27 15:24:26.910592024 +0100
@@ -37,6 +37,30 @@
 #define WEBRTC_ARCH_32_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
 #define WEBRTC_LITTLE_ENDIAN
+#elif defined(__powerpc64__)
+#define WEBRTC_ARCH_PPC_FAMILY
+#define WEBRTC_ARCH_PPC
+#define WEBRTC_ARCH_64_BITS
+#define WEBRTC_ARCH_BIG_ENDIAN
+#define WEBRTC_BIG_ENDIAN
+#elif defined(__ppc__) || defined(__powerpc__) || defined(__PPC__)
+#define WEBRTC_ARCH_PPC_FAMILY
+#define WEBRTC_ARCH_PPC
+#define WEBRTC_ARCH_32_BITS
+#define WEBRTC_ARCH_BIG_ENDIAN
+#define WEBRTC_BIG_ENDIAN
+#elif defined(__s390x__)
+#define WEBRTC_ARCH_S390_FAMILY
+#define WEBRTC_ARCH_S390
+#define WEBRTC_ARCH_64_BITS
+#define WEBRTC_ARCH_BIG_ENDIAN
+#define WEBRTC_BIG_ENDIAN
+#elif defined(__s390__)
+#define WEBRTC_ARCH_S390_FAMILY
+#define WEBRTC_ARCH_S390
+#define WEBRTC_ARCH_32_BITS
+#define WEBRTC_ARCH_BIG_ENDIAN
+#define WEBRTC_BIG_ENDIAN
 #elif defined(__ARMEL__)
 // TODO(andrew): We'd prefer to control platform defines here, but this is
 // currently provided by the Android makefiles. Commented to avoid duplicate
